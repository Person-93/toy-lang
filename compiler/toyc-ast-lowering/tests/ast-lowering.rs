use test_util::{assert_debug_snapshot, snapshots, TestSession};
use toyc_ast_lowering::HirContext;

fn main() {
  snapshots(|text| {
    let session = TestSession::new("test-ast-lowering");
    let ast = toyc_ast::parse_single_file(text, &session.handler).unwrap();
    let ctx = HirContext::new(&session);
    assert_debug_snapshot!(ctx.lower_package(&ast));
  });
}
