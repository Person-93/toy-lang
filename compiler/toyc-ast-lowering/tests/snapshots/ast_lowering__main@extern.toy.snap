---
source: compiler/toyc-ast-lowering/tests/ast-lowering.rs
assertion_line: 9
expression: ctx.lower_package(&ast)

---
Package {
    attrs: [],
    items: [
        Function(
            Function {
                id: HirId(
                    1,
                    PhantomData,
                    PhantomData,
                ),
                ident: answer @ 14..20,
                visibility: Inherited,
                fn_type: FnType {
                    extern_: Yes(
                        C,
                    ),
                    unsafety: No,
                    constness: No,
                    generics: None,
                    inputs: [],
                    output: Explicit(
                        Type {
                            id: HirId(
                                2,
                                PhantomData,
                                PhantomData,
                            ),
                            kind: Primitive(
                                Int(
                                    Eight,
                                ),
                            ),
                            span: Span {
                                lo: 26,
                                hi: 28,
                            },
                        },
                    ),
                    self_kind: None,
                },
                body_id: Some(
                    BodyId(
                        HirId(
                            5,
                            PhantomData,
                            PhantomData,
                        ),
                    ),
                ),
                span: Span {
                    lo: 0,
                    hi: 36,
                },
            },
        ),
        Function(
            Function {
                id: HirId(
                    6,
                    PhantomData,
                    PhantomData,
                ),
                ident: ping @ 52..56,
                visibility: Inherited,
                fn_type: FnType {
                    extern_: Yes(
                        C,
                    ),
                    unsafety: No,
                    constness: No,
                    generics: None,
                    inputs: [],
                    output: Explicit(
                        Type {
                            id: HirId(
                                7,
                                PhantomData,
                                PhantomData,
                            ),
                            kind: Primitive(
                                Uint(
                                    Size,
                                ),
                            ),
                            span: Span {
                                lo: 62,
                                hi: 67,
                            },
                        },
                    ),
                    self_kind: None,
                },
                body_id: Some(
                    BodyId(
                        HirId(
                            10,
                            PhantomData,
                            PhantomData,
                        ),
                    ),
                ),
                span: Span {
                    lo: 38,
                    hi: 79,
                },
            },
        ),
        Function(
            Function {
                id: HirId(
                    11,
                    PhantomData,
                    PhantomData,
                ),
                ident: pong @ 95..99,
                visibility: Inherited,
                fn_type: FnType {
                    extern_: Yes(
                        C,
                    ),
                    unsafety: No,
                    constness: No,
                    generics: None,
                    inputs: [],
                    output: Explicit(
                        Type {
                            id: HirId(
                                12,
                                PhantomData,
                                PhantomData,
                            ),
                            kind: Primitive(
                                Uint(
                                    Size,
                                ),
                            ),
                            span: Span {
                                lo: 105,
                                hi: 110,
                            },
                        },
                    ),
                    self_kind: None,
                },
                body_id: None,
                span: Span {
                    lo: 81,
                    hi: 111,
                },
            },
        ),
    ],
    span: Span {
        lo: 0,
        hi: 111,
    },
}
